### PR TITLE
[FCE-306] Remove metadata from public api

### PR DESCRIPTION
**Pull request by kamil-stasiak**
## Description

- PeerMetadata and TrackMetadata generics from the react-client SDK were replaced by ordinary types.
- PeerMetadata and TrackMetadata generics from the ts-client SDK were kept.
- Fixed metadata update after `addTrack`.

## Motivation and Context

We don't want to allow the user to change or see peer and track metadata. We have decided to use it only as an internal mechanism.

## Types of changes

Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: packages/react-client/src/Client.ts [hunk 1]
```diff
@@ -718,9 +688,11 @@ export class Client<PeerMetadata, TrackMetadata> extends (EventEmitter as {
 
   public getTsClient = () => this.tsClient;
 
-  public connect(config: ConnectConfig<PeerMetadata>): void {
+  public connect(config: ConnectConfig): void {
```
[the remark this answers is not in the document]
Comment (kamil-stasiak) [reply packages/react-client/src/Client.ts:675]: It's mandatory because there is a Fishjam URL and a peer token

File: packages/react-client/src/Client.ts [hunk 2]
```diff
@@ -718,9 +688,11 @@ export class Client<PeerMetadata, TrackMetadata> extends (EventEmitter as {
...
     this.status = "connecting";
-    this.tsClient.connect(config);
+    this.audioTrackManager.setDisplayName(config?.peerMetadata?.displayName);
+    this.videoTrackManager.setDisplayName(config?.peerMetadata?.displayName);
+    this.tsClient.connect({ ...config, peerMetadata: config?.peerMetadata ?? {} });
```
[the remark this answers is not in the document]
Comment (kamil-stasiak) [reply packages/react-client/src/Client.ts]: I removed this part of a code because I decided to implement it in a separate task

File: packages/react-client/src/screenShareTrackManager.ts [hunk 3]
```diff
@@ -10,26 +10,26 @@ const getTracks = (stream: MediaStream): { video: MediaStreamTrack; audio: Media
...
+  const startStreaming: ScreenshareApi["startStreaming"] = async (props) => {
     const stream = await navigator.mediaDevices.getDisplayMedia({
       video: props?.videoConstraints ?? true,
       audio: props?.audioConstraints ?? true,
     });
     const { video, audio } = getTracks(stream);
 
-    const addTrackPromises = [tsClient.addTrack(video, props?.metadata)];
-    if (audio) addTrackPromises.push(tsClient.addTrack(audio, props?.metadata));
+    const addTrackPromises = [tsClient.addTrack(video)];
```
[the remark this answers is not in the document]
Comment (kamil-stasiak) [reply packages/react-client/src/screenShareTrackManager.ts:25]: https://linear.app/swmansion/issue/FCE-417/pass-displayname-as-a-internal-trackmetadata

File: packages/ts-client/src/webrtc/webRTCEndpoint.ts [hunk 4]
```diff
@@ -326,14 +326,13 @@ export class WebRTCEndpoint<EndpointMetadata = any, TrackMetadata = any> extends
...
-    Object.values(data.midToTrackId)
+    Object.keys(data.midToTrackId)
       .map((trackId) => {
         if (!trackId) throw new Error('TrackId is not defined');
-        if (typeof trackId !== 'string') throw new Error('TrackId is not a string');
 
         return trackId;
       })
-      .map((trackId) => this.local.getTrackByMidOrNull(trackId))
+      .map((mid) => this.local.getTrackByMidOrNull(mid))
```
Comment (mironiasty) [inline packages/ts-client/src/webrtc/webRTCEndpoint.ts:335]: what does `mid` mean?

File: packages/ts-client/src/webrtc/webRTCEndpoint.ts [hunk 5]
```diff
@@ -326,14 +326,13 @@ export class WebRTCEndpoint<EndpointMetadata = any, TrackMetadata = any> extends
     this.remote.updateMLineIds(data.midToTrackId);
     this.local.updateMLineIds(data.midToTrackId);
 
-    Object.values(data.midToTrackId)
+    Object.keys(data.midToTrackId)
```
Comment (mironiasty) [inline packages/ts-client/src/webrtc/webRTCEndpoint.ts:329]: Was this change required because of other changes in this PR, or is it separate bug?

File: packages/react-client/src/useSetupMedia.ts [hunk 6]
```diff
@@ -51,7 +46,7 @@ export const createUseSetupMediaHook = <PeerMetadata, TrackMetadata>(
             pending = true;
 
             await videoTrackManager
-              .startStreaming(config.defaultTrackMetadata, config.defaultSimulcastConfig, config.defaultMaxBandwidth)
+              .startStreaming(config.defaultSimulcastConfig, config.defaultMaxBandwidth)
               .finally(() => {
                 pending = false;
               });
```
Comment (mironiasty) [inline packages/react-client/src/useSetupMedia.ts:52]: I don't think we should mix `thenable` syntax with `awaitable` 🤔 
```suggestion
            try {
              await videoTrackManager.startStreaming(config.defaultSimulcastConfig, config.defaultMaxBandwidth);
            } finally {
              pending = false;
            }
```
(this applies to other places in this file too - so maybe better to fix it in separate PR)

File: packages/react-client/src/trackManager.ts [hunk 7]
```diff
@@ -48,8 +53,24 @@ export class TrackManager implements GenericTrackManager {
...
+    mediaManager?.start(deviceId ?? true);
+  }
+
+  async function stop() {
+    mediaManager?.stop();
+  }
+
+  function getType(): TrackMetadata["type"] {
+    return TRACK_TYPE_TO_DEVICE[mediaManager.getDeviceType()];
+  }
```
Comment (czerwiukk) [inline packages/react-client/src/trackManager.ts:70]: **nitpick:** since this is a react hook, we could just do
```suggestion
  const deviceType = TRACK_TYPE_TO_DEVICE[mediaManager.getDeviceType()]
```

File: packages/react-client/src/trackManager.ts [hunk 8]
```diff
@@ -80,9 +83,12 @@ export const useTrackManager = <PeerMetadata, TrackMetadata>({
     // see `getRemoteOrLocalTrackContext()` explanation
     setCurrentTrackId(media.track.id);
 
+    const trackMetadata = { type: getType(), paused: false };
```
Comment (mironiasty) [inline packages/react-client/src/trackManager.ts:86]: WDYT about having some type for such variables, so we can use it for easier refactoring in future?

File: packages/react-client/src/trackManager.ts [hunk 9]
```diff
@@ -62,11 +65,11 @@ export const useTrackManager = <PeerMetadata, TrackMetadata>({
...
   }
 
-  async function startStreaming(
-    trackMetadata?: TrackMetadata,
-    simulcastConfig?: SimulcastConfig,
-    maxBandwidth?: TrackBandwidthLimit,
-  ) {
+  function getType(): TrackMetadata["type"] {
+    return TRACK_TYPE_TO_DEVICE[mediaManager.getDeviceType()];
+  }
```
Comment (mironiasty) [inline packages/react-client/src/trackManager.ts:70]: would it be simpler to write just if/switch statement here?

File: packages/react-client/src/index.ts [hunk 10]
```diff
@@ -47,7 +49,7 @@ export type {
   VadStatus,
   EncodingReason,
   MetadataParser,
-  ConnectConfig,
+  ConnectConfig as TSClientConnectConfig,
```
Comment (mironiasty) [inline packages/react-client/src/index.ts:52]: Do we want to export `TSClientConnectConfig` here, or version modified in one of flies below?

File: packages/ts-client/src/webrtc/webRTCEndpoint.ts (hunk 4 above)
Comment (kamil-stasiak) [reply packages/ts-client/src/webrtc/webRTCEndpoint.ts:335]: In Session Description Protocol (SDP) it's `media identifier`, a unique identifier for media stream

Comment (czerwiukk) [reply packages/ts-client/src/webrtc/webRTCEndpoint.ts:335]: i feel `mediaId` would give me a better understanding what this variable contains, hbu?

File: packages/react-client/src/trackManager.ts (hunk 9 above)
Comment (kamil-stasiak) [reply packages/react-client/src/trackManager.ts:70]: I've implemented this, as @czerwiukk said in [this](https://github.com/fishjam-cloud/web-client-sdk/pull/69#discussion_r1731092311) comment. I don't have any preferences. Map is okej for me.

File: packages/react-client/src/index.ts (hunk 10 above)
Comment (kamil-stasiak) [reply packages/react-client/src/index.ts]: The new one. Fixed

File: packages/react-client/src/trackManager.ts (hunk 8 above)
Comment (kamil-stasiak) [reply packages/react-client/src/trackManager.ts]: Refactored as @czerwiukk said[here](https://github.com/fishjam-cloud/web-client-sdk/pull/69#discussion_r1731092311 comment)

File: packages/ts-client/src/webrtc/webRTCEndpoint.ts (hunk 4 above)
Comment (kamil-stasiak) [reply packages/ts-client/src/webrtc/webRTCEndpoint.ts:335]: `mid` comes from RFC https://www.rfc-editor.org/rfc/rfc5888.html. I'm not sure if this change will introduce more confusion

Comment (mironiasty) [reply packages/ts-client/src/webrtc/webRTCEndpoint.ts:335]: In that case - should it be updated in all `.map` functions here? above one is returning `trackId`

**Review by mironiasty — APPROVED** [review]
One more comment for naming. Few other things could be still improved in incoming PRs

File: packages/react-client/src/state.types.ts [hunk 11]
```diff
@@ -19,24 +19,24 @@ export type Track<TrackMetadata> = {
   track: MediaStreamTrack | null;
 };
 
-export interface Origin<OriginMetadata> {
+export interface Origin {
   id: string;
   type: string;
-  metadata?: OriginMetadata;
+  metadata?: PeerMetadata;
   rawMetadata: any; // eslint-disable-line @typescript-eslint/no-explicit-any
```
[the remark this answers is not in the document]
Comment (kamil-stasiak) [reply packages/react-client/src/state.types.ts]: Changed

File: packages/react-client/src/trackManager.ts [hunk 12]
```diff
@@ -1,30 +1,42 @@
...
   private currentTrackId: string | null = null;
-
-  constructor(tsClient: FishjamClient<PeerMetadata, TrackMetadata>, deviceManager: GenericMediaManager) {
+  private paused: boolean = false;
+  private displayName?: string;
+
+  constructor(
+    tsClient: FishjamClient<PeerMetadata, TrackMetadata>,
+    deviceManager: GenericMediaManager,
+    type: TrackMetadata["type"],
```
[the remark this answers is not in the document]
Comment (kamil-stasiak) [reply packages/react-client/src/trackManager.ts]: Fixed

**Review by czerwiukk — APPROVED** [review]
